uint64 ns_since_epoch
uint64 thread_id
string label
